flexget: 1.2.337 -> 2.8.17

tari opened the pull request:
###### Motivation for this change

Flexget 1.2.337 isn't as old as it sounds (released July 2015) but this update does away with some dependencies on obsolete library versions.

Flexget's state does not upgrade from the old version successfully, so upgrading requires some manual intervention.

###### Things done

- [ ] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [x] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

FRidh commented:
Thanks! Please put each package expression you touch in a separate commit instead one big commit like you have now.

FRidh commented on pkgs/applications/networking/flexget/default.nix:
```diff
@@ -9,23 +9,30 @@
+  checkPhase = ''
+    py.test
+  '';
 
```
maybe `py.test flextest/tests/api_tests` since that seems to be the only folder with tests.

tari replied:
api_tests is the problematic directory:

```
flexget/tests/api_tests/test_server_api.py:13: in <module>
    from flexget.tests.conftest import MockManager
E   ImportError: No module named tests.conftest
```

Which in turn appears to make all of the tests in that directory fail with `fixture 'api_client' not found`.

FRidh replied:
It seems a significant part of the test suite is not included
https://github.com/Flexget/Flexget/tree/develop/flexget/tests
You might want to fetch from GitHub instead of PyPI.

tari replied:
Oh, interesting. I was referring to sources on github and didn't even consider there might be items missing from the pypi distribution. It works a little better when fetched from github, but requires additional (unpackaged) modules so I'll leave the tests disabled for now.

FRidh replied:
That is up to you as maintainer. Just remember, if there aren't any tests, those that update dependencies of flexget won't know whether flexget still works. And that happened recently :)

FRidh commented on pkgs/applications/networking/flexget/default.nix:
```diff
@@ -36,4 +43,4 @@ buildPythonPackage rec {
     license = lib.licenses.mit;
     maintainers = with lib.maintainers; [ domenkozar ];
```
would you like to maintain this package?

FRidh reviewed (changes requested):
As noted above, please split it in multiple commits.

Did you check whether tests are run for all packages with all interpreter versions?

This looks good and from the changes made I don't think this will break anything.

tari committed:
aniso8601: init at 1.2.0

tari committed:
apscheduler: 3.0.4 -> 3.3.1

tari committed:
babelfish: 0.5.3 -> 0.5.5

tari committed:
cherrypy: 3.2.2 -> 8.7.0

tari committed:
flask-compress: init at 1.3.2

tari committed:
flask-restful: init at 0.3.5

tari committed:
flask-restplus: init at 0.8.6

tari committed:
rebulk: init at 0.8.2

tari committed:
guessit: 0.9.4 -> 2.0.4

tari committed:
colorclass: init at 2.2.0

tari committed:
safe: init at 0.4

tari committed:
terminaltables: init at 3.1.0

tari committed:
flexget: 1.2.337 -> 2.8.17

tari commented:
> As noted above, please split it in multiple commits.

Done.

> Did you check whether tests are run for all packages with all interpreter versions?

Is there a recommended non-manual way to do that?

tari commented:
>> Did you check whether tests are run for all packages with all interpreter versions?
>
> Is there a recommended non-manual way to do that?

I ran `nox-review pr 21636` and made a few changes relating to py3k where the packages were broken.

Everything appears okay except zope_configuration and the packages that depend on it because of #20791. Most relevant to this PR among packages affected by #20791 is WSME which depends on cherrypy which was updated.

Mic92 commented:
Maybe the necessary manual intervention could be added to the changelog.

tari commented:
Added an item to release notes.

@FRidh anything else to be done? I think I've addressed all of your comments.

joachifm commented:
Needs a rebase. @FRidh is this ready to integrate otherwise?

FRidh commented:
@joachifm I guess so. I do like to do a `nox-review pr` first but that requires the rebase.

tari commented:
Rebased.

There are more packages to add so the tests pass, but I'll plan to do that in another PR.

FRidh commented:
> There are more packages to add so the tests pass, but I'll plan to do that in another PR.

If you maintain this package, then that's fine. But please do so before we branch off for 17.03.
I got the following error when I ran `$ nox-review pr 21636`
```
No local packages or working download links found for setuptools_scm
Traceback (most recent call last):
  File "nix_run_setup.py", line 8, in <module>
    exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\\r\\n', '\\n'), __file__, 'exec'))
  File "setup.py", line 89, in <module>
    'zookeeper = apscheduler.jobstores.zookeeper:ZookeeperJobStore [zookeeper]'
  File "/nix/store/wmbq6d0lk01s28jy2sgy7ir2qlkb1zkp-python-2.7.13/lib/python2.7/distutils/core.py", line 111, in setup
    _setup_distribution = dist = klass(attrs)
  File "/nix/store/191knf3y8c92ahlcj9ziry1idfckzaz0-python2.7-bootstrapped-pip-9.0.1/lib/python2.7/site-packages/setuptools/dist.py", line 315, in __init__
    self.fetch_build_eggs(attrs['setup_requires'])
  File "/nix/store/191knf3y8c92ahlcj9ziry1idfckzaz0-python2.7-bootstrapped-pip-9.0.1/lib/python2.7/site-packages/setuptools/dist.py", line 361, in fetch_build_eggs
    replace_conflicting=True,
  File "/nix/store/191knf3y8c92ahlcj9ziry1idfckzaz0-python2.7-bootstrapped-pip-9.0.1/lib/python2.7/site-packages/pkg_resources/__init__.py", line 846, in resolve
    dist = best[req.key] = env.best_match(req, ws, installer)
  File "/nix/store/191knf3y8c92ahlcj9ziry1idfckzaz0-python2.7-bootstrapped-pip-9.0.1/lib/python2.7/site-packages/pkg_resources/__init__.py", line 1118, in best_match
    return self.obtain(req, installer)
  File "/nix/store/191knf3y8c92ahlcj9ziry1idfckzaz0-python2.7-bootstrapped-pip-9.0.1/lib/python2.7/site-packages/pkg_resources/__init__.py", line 1130, in obtain
    return installer(requirement)
  File "/nix/store/191knf3y8c92ahlcj9ziry1idfckzaz0-python2.7-bootstrapped-pip-9.0.1/lib/python2.7/site-packages/setuptools/dist.py", line 429, in fetch_build_egg
    return cmd.easy_install(req)
  File "/nix/store/191knf3y8c92ahlcj9ziry1idfckzaz0-python2.7-bootstrapped-pip-9.0.1/lib/python2.7/site-packages/setuptools/command/easy_install.py", line 659, in easy_install
    raise DistutilsError(msg)
distutils.errors.DistutilsError: Could not find suitable distribution for Requirement.parse('setuptools_scm')
builder for ‘/nix/store/1nqzkihd5jnqpjf3310arnxb7c8lh73b-python2.7-APScheduler-3.3.1.drv’ failed with exit code 1
killing process 21303
error: build of ‘/nix/store/1nqzkihd5jnqpjf3310arnxb7c8lh73b-python2.7-APScheduler-3.3.1.drv’ failed
```


@domenkozar you're listed also as maintainer. Will you still?

tari commented:
I can't reproduce that build failure for apscheduler- it succeeds for me both on NixOS and a fresh Ubuntu virtual machine with Nix, though setuptools_scm is clearly declared as a setup dependency so it's not clear to me why it's succeeding here.

In any case, I added setuptools_scm as a dependency which should fix that- it (still) builds for me.

FRidh commented:
@tari enable the sandbox and try again :)

> Tested using sandboxing
(nix.useSandbox on NixOS,
or option build-use-sandbox in nix.conf
on non-NixOS)

tari committed:
zerobin: disable tests because it doesn't have any

It does however contain a copy of cherrypy that doesn't get installed,
which fails tests when it tries to import from cherrypy and gets imports
from the version provided by Nix (which is probably not the same one as
is having its tests run).

tari commented:
Got it; I misunderstood the docs as to the extent of the effect of sandboxing.

It's a little difficult to tell with the unrelated zope build failures, but I think I've made everything work correctly that is expected to, with the exception of python35Packages.zerobin which fails running tests.

That appears to be a bug in how zerobin is packaged though, since the source distribution appears to include a copy of cherrypy that does not get installed, but is picked up when running tests and in turn tries to import modules from the version of cherrypy provided by Nix (which is not the same version).

Meanwhile, zerobin itself doesn't appear to have any actual tests so I added a commit that disables tests for it since that wasn't doing anything useful.

tari committed:
httpbin: 0.2.0 -> 0.5.0

tari committed:
rednose: init at 1.2.1

tari committed:
pytest-httpbin: init at 0.2.3

tari committed:
vcrpy: init at 1.10.5

tari committed:
flask-restplus: enable tests

tari committed:
flexget: enable tests

tari committed:
release notes: incompatible flexget upgrade

tari commented:
Rebased and added dependencies so flexget can be tested. As best I can tell (with sandboxing), everything works.

7c6f434c commented:
I am trying to triage the moderately old PRs, because the backlog has got out of hand again.

Let me just admit that I have failed to resolve the conflicts quickly, but I did push some safe additions/updates (basically as much as nox-review didn't complain about)

FRidh commented:
still flexget, I thought we were already at the latest version. I'll check out this PR.

FRidh commented:
The latest version of flexget is available at https://github.com/NixOS/nixpkgs/commit/4f7421331911799a39e0f77ffeb2d11acaeff292.
